File: academic-system-python/src/academic_system/controller/authentication_controller.py
from academic_system.security.authentication_service import AuthenticationService
from academic_system.exception.exceptions import AuthenticationException
from academic_system.model.user import User


class AuthenticationController:
    def __init__(self):
        self._auth_service = AuthenticationService()

    def authenticate(self, username: str, password: str) -> bool:
        try:
            self._auth_service.authenticate(username, password)
            return True
        except AuthenticationException:
            return False

    def authenticate_and_return(self, username: str,
                                password: str) -> User:
        return self._auth_service.authenticate(username, password)

File: academic-system/src/main/java/org/example/academic/system/controller/AuthenticationController.java
package org.example.academic.system.controller;

import org.example.academic.system.exception.AuthenticationException;
import org.example.academic.system.model.User;
import org.example.academic.system.security.AuthenticationService;

public class AuthenticationController {

    private final AuthenticationService authService;

    public AuthenticationController() {
        this.authService = new AuthenticationService();
    }

    public boolean authenticate(String username, String password) {
        try {
            User user = authService.authenticate(username, password);
            return user != null;
        } catch (AuthenticationException e) {
            return false;
        }
    }
}